commands/board/list.go

```diff
@@ -18,44 +18,50 @@
 package board
 
 import (
-	"context"
-	"errors"
-	"fmt"
-
 	"github.com/arduino/arduino-cli/commands"
 	rpc "github.com/arduino/arduino-cli/rpc/commands"
+	"github.com/pkg/errors"
 )
 
 // List FIXMEDOC
-func List(ctx context.Context, req *rpc.BoardListReq) (*rpc.BoardListResp, error) {
-	pm := commands.GetPackageManager(req)
+func List(instanceID int32) (*rpc.BoardListResp, error) {
+	pm := commands.GetPackageManager(instanceID)
 	if pm == nil {
 		return nil, errors.New("invalid instance")
 	}
 
+	serialDiscovery, err := commands.NewBuiltinSerialDiscovery(pm)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to instance serial-discovery")
+	}
+
+	if err := serialDiscovery.Start(); err != nil {
+		return nil, errors.Wrap(err, "unable to start serial-discovery")
+	}
+	defer serialDiscovery.Close()
+
 	resp := &rpc.BoardListResp{Ports: []*rpc.DetectedPort{}}
-	for _, disc := range commands.GetDiscoveries(req) {
```

Comment:
instead of going through a dynamic loading mechanism that will always return just the serial discovery, we use it directly